src/dir.cr

```diff
@@ -93,6 +96,24 @@ class Dir
     String.new(dir).tap { C.free(dir as Void*) }
   end
 
+  def self.chdir path
+    if C.chdir(path) != 0
+      raise Errno.new("Error while changing directory")
+    end
+  end
+
+  def self.chdir(path)
+    old = working_directory
+    chdir(path)
+    yield
```

Comment:
I'm confused. Isn't there two methods with the same signature here? L99 and L105. 
If there's no block passed what does yield do? I'm a crystal newbie :-)